### PR TITLE
start server promise and careful way

### PR DESCRIPTION
**Pull request by jupe**
## Status
**IN DEVELOPMENT**

## Migrations
NO

## Description
Starting server more carefully using Promise flow. 
* Update mongoose-query -dependency
* Wait until mongo connection is done before continue - if fails shutdown whole server immediately.

## Todos
- [ ] Tests

### CODE REVIEW

File: app/db.js [hunk 1]
```diff
@@ -23,14 +23,16 @@ const connect = function () {
...
+      });
+    });
 };
-
-mongoose.connection.on('error', () => {
-  logger.error(`Could not connect to MongoDB: ${dbUrl}`);
-});
-
-mongoose.connection.on('disconnected', () => {
+/* mongoose.connection.on('disconnected', () => {
```
Comment (jupe) [inline app/db.js]: this probably should reactivated

File: app/index.js [hunk 2]
```diff
@@ -54,25 +54,20 @@ const server = Server(app);
...
-
-// Bootstrap addons, like default webGUI
-AddonManager.init(app, server, io);
-
-AddonManager.loadAddons().then(() => {
-  AddonManager.registerAddons().then(() => {
+DB.connect().catch((error) => {
+  logger.error('mongoDB connection failed: ', error.message);
+  process.exit(-1);
+}).then(() => models.registerModels())
```
Comment (jupe) [inline app/index.js]: comments is taken away but I think function names are enough descriptive..

**Comment by jupe** [comment]
This would be needed for clustering approach..

File: app/index.js (hunk 2 above)
Comment (olkorhon) [inline app/index.js]: Agreed, the name is quite self-evident, to me at least.

File: app/db.js (hunk 1 above)
Comment (olkorhon) [inline app/db.js]: is there a reason why you disabled disconnect and reconnect event handling?

File: app/index.js [hunk 3]
```diff
@@ -54,25 +54,20 @@ const server = Server(app);
...
+  .then(() => AddonManager.loadAddons())
+  .then(() => AddonManager.registerAddons())
+  .then(() =>
     // Error route should be initialized after addonmanager has served all static routes
-    routes.registerErrorRoute(app);
-
+    routes.registerErrorRoute(app)
+  )
+  .then(() => {
     function onError(error) {
```
Comment (olkorhon) [inline app/index.js]: SIGINT is also handled by master.js, 2 separate handlers might cause issues.

**Comment by jupe** [comment]
Maybe we can close this one because cluster branch contains same changes..

**Pull request closed without merging**